Documentation/Contributing.md

```diff
@@ -1,75 +1,49 @@
-# Development
+# Contributing to Swift Package Manager
+There are several types of contributions one can make. Bug fixes, documentation and enhancements that do not materially change the user facing semantics of Swift Package Manager should be submitted directly as PR.  
 
-This document contains information on building and testing the Swift Package
-Manager. There are many ways to develop Swift Package Manager. The official way
-is to use Swift's build-script which builds the full compiler toolchain but
-that's rarely required.
+Larger changes that do materially change the semantics of Swift Package Manager (e.g. changes to the manifest format or behavior) are required to go through [Swift Evolution Process](https://github.com/apple/swift-evolution/blob/master/process.md).  
 
-## Using the Swift Compiler Build Script
+To see how previous evolution decisions for SwiftPM have been made and have some direction for the development of future features please check out the [Community Proposal](Documentation/Internals/PackageManagerCommunityProposal.md).  
 
-Follow [these](https://github.com/apple/swift#getting-started) instructions to
-get the Swift sources and then execute the `build-script` using swiftpm preset:
+For more information about making contributions to the Swift project in general see [Swift Contribution Guide](https://swift.org/contributing/)  
 
-### macOS
+## Reporting issues
+Report a bug guide: https://github.com/apple/swift-package-manager/blob/main/Documentation/Resources.md#reporting-a-good-swiftpm-bug.  
+JIRA Bug Tracker (a place where you can open bugs, enhancements to start to contribute): [https://bugs.swift.org/browse/SR-13640?jql=component%20%3D%20%22Package%20Manager%22](https://bugs.swift.org/browse/SR-13640?jql=component%20%3D%20%22Package%20Manager%22).
 
-```sh
-$ ./swift/utils/build-script --preset=buildbot_swiftpm_macos_platform,tools=RA,stdlib=RA
-```
+## Development environment
+If you are contributing using macOS, the best option is to use Xcode to build and run test SwiftPM.  
+You have multiple ways to setup your development environment, here we will focus on two:  
+* [Using Xcode](#using-xcode)
+* [Using a standalone Swift toolchain](#using-standalone)
+* [Using a Trunk Snapshot](#using-trunk-snapshot)
+* [Self Hosting](#self-hosting)
+* [Using the Swift Compiler Build Script](#swift-compiler-build-script)
```

Comment:
This section needs further work. There is no material difference between "Using a standalone Swift toolchain", "Using a Trunk Snapshot" and "Self Hosting". I suggest the following text:


```markdown
##Development environment

SwiftPM is typically built with a pre-existing versions of SwiftPM present on the system. Which can be obtained by installing the latest release of Xcode or a toolchain from swift.org. But there are have multiple ways to setup your development environment:

### Using Xcode (Easiest)

1. Install latest Xcode from https://developer.apple.com/xcode
2. Open SwiftPM's `Package.swift` manifest with the latest release (including betas) of Xcode.

### Using a Swift toolchain

1. Download a toolchain from https://swift.org/download/

2. Confirm the toolchain is installed correctly:

macOS
$ export TOOLCHAINS=swift
# Verify that we're able to find the swift compiler from the installed toolchain.
$ xcrun --find swift
/Library/Developer/Toolchains/swift-latest.xctoolchain/usr/bin/swift

Linux
$ export PATH=/path/to/swift-toolchain/usr/bin:"${PATH}"
# Verify that we're able to find the swift compiler from the installed toolchain.
$ which swift
/path/to/swift-toolchain/usr/bin/swift

3. Alternatively use tools like swiftenv that help manage toolchains versions.

### Using the Swift Compiler Build Script

Follow these instructions to get the Swift sources and then execute the build-script using swiftpm preset:

macOS
$ ./swift/utils/build-script --preset=buildbot_swiftpm_macos_platform,tools=RA,stdlib=RA

Linux
$ ./swift/utils/build-script --preset=buildbot_swiftpm_linux_platform,tools=RA,stdlib=RA
Once the build is complete, you should be able to run the swiftpm binaries from the build folder.

Example

$ cd /tmp && mkdir hello && cd hello
$ /path/to/swiftpm/.build/x86_64-apple-macosx/debug/swift-package init
$ /path/to/swiftpm/.build/x86_64-apple-macosx/debug/swift-build
$ Utilities/bootstrap test
```